certbot/tests/compat/filesystem_test.py

```diff
@@ -6,41 +6,162 @@
 from certbot.tests.util import TempDirTestCase
 
 
-class FilesystemTest(TempDirTestCase):
-    """Unit tests for filesystem module."""
-    @unittest.skipIf(os.name != 'nt', reason='Test specific to Windows security')
+@unittest.skipIf(os.name != 'nt', reason='Test specific to Windows security')
+class WindowsChmodTests(TempDirTestCase):
+    """Unit tests for Windows chmod function in filesystem module"""
+    def test_symlink_resolution(self):
+        probe_path = _create_probe(self.tempdir)  # This is 0o744 by default
+
+        link_path = os.path.join(self.tempdir, 'link')
+        os.symlink(probe_path, link_path)
+
+        ref_dacl_probe = _get_security_dacl(probe_path).GetSecurityDescriptorDacl()
+        ref_dacl_link = _get_security_dacl(link_path).GetSecurityDescriptorDacl()
+
+        # Removing the rights for `all`, we have at least one ACL less than in the case of 0o744.
+        filesystem.chmod(link_path, 0o700)
+
+        # Assert the real file is impacted, not the link.
+        cur_dacl_probe = _get_security_dacl(probe_path).GetSecurityDescriptorDacl()
+        cur_dacl_link = _get_security_dacl(link_path).GetSecurityDescriptorDacl()
+        self.assertFalse(filesystem._compare_dacls(ref_dacl_probe, cur_dacl_probe))  # pylint: disable=protected-access
+        self.assertTrue(filesystem._compare_dacls(ref_dacl_link, cur_dacl_link))  # pylint: disable=protected-access
+
+    def test_world_permission(self):
+        import win32security  # pylint: disable=import-error
+        probe_path = _create_probe(self.tempdir)
+
+        everybody = win32security.ConvertStringSidToSid('S-1-1-0')
```

Comment:
What do you think about making global constants in `filesystem.py` for these magic strings? Alternatively we could make functions to return the specific SID objects we're interested in.

Reply:
Done. As global constants.